### ISSUE TITLE
Phone locks up every 20 mins

### ISSUE THREAD

**Issue by jethrogb** [ukanth/afwall#794]
With AFWall+ enabled, my phone completely locks up after 20 minutes, then automatically reboots. If I have AFWall+ disabled, there is no problem. If I enable AFWall+ after the system is running, the problem happens 20 minutes after enabling AFWall+.

Other than this, AFWall+ seems to be working fine.

I'm using a Moto X4 with Android 8.0.0, kernel 4.4.78. I'm not seeing any useful info in logcat (probably because the crash happens before logging output can be written).

**Comment by ildar** [comment]
You may want to look deeper with adb logcat. In your case with USB cable.

**Comment by jethrogb** [comment]
I edited my post, you might not have seen the change since you're replying by e-mail:

> I'm not seeing any useful info in logcat (probably because the crash happens before logging output can be written).

**Comment by ildar** [comment]
ah! indeed, didn't see.
Do you see AFWall getting root short before crash?
BTW, system crashes produce some extended info, "dontpanic", "apanic" or
tombstones. May help.

**Comment by ukanth** [comment]
Disable "Startup Delay" if you have enabled it. !

**Comment by duckenfeld** [comment]
Hi. I have the same issue (30 seconds till 3 minutes after starting the phone) on a Moto Z2 force too. Startup Delay is not enabled.

**Comment by ukanth** [comment]
Need logcat in that case. Is log service enabled ?

**Comment by duckenfeld** [comment]
I think you mean this? 

That are the logs after start and lock 

AppID :	10007
App-Name:	Motorola
Gesamt blockierte Pakete:	1
[TCP]184.72.223.191:443(1)

	---------
AppID :	10146
App-Name:	Official TWRP App
Gesamt blockierte Pakete:	1
[TCP]52.19.109.206:443(1)

	---------
AppID :	10198
App-Name:	Yatse
Gesamt blockierte Pakete:	1
[TCP]54.225.233.150:443(1)

	---------

**Comment by ximmox** [comment]
Hello, 

I am experiencing the same problem, there are **seemingly random lockups resulting in an automatic reboot.** (it doesn't seem to be a TOTAL reboot, as I don't have to input my PIN again afterwards, it's bootlogo, lineage bootanimation and back to the lockscreen).

Device: **Moto Z** (there might be a pattern here, all 3 reports are Motorola devices. @duckenfeld: is yours already on Android 8 Oreo, too?)
ROM: LineageOS 15.1-20180430 with microG (= Android 8.1)
Kernel: 3.18.71-LineageOS
AFWall v2.9.9 (startup delay is disabled)

These reboots did **not** occur on LineageOS 14.1 (Android 7.1) I have been using until yesterday.

The lockups seem random, but are mostly occuring with heavy network traffic (reloading many tabs in firefox at once, opening apps shortly after each other that update/sync content)
No difference wether DNSproxy is on Auto or netd disabled.

I'll try to get the necessary logs when I am at home with my laptop.

**Comment by duckenfeld** [comment]
Hi,

my device is a Moto Z2 force, with Oreo 8.0,

Kind regards,
duckenfeld

**Comment by ximmox** [comment]
Thanks for the response @duckenfeld.

My promise of delivering logs wasn't forgotten, but I tried for 20 minuten today to make my Moto Z fail & reboot and "sadly" it was stable.
I will see if the error shows up again tomorrow, I will leave AFWall enabled for now.

(One thing that MIGHT have an impact: Android 8 has a hidden setting in the developer settings which keeps the mobile data connection always on even though WIFI is connected (see here: https://www.androidpolice.com/2017/12/09/fyi-mobile-data-now-set-always-active-default-oreo/). This was enabled by default on my Lineage OS 15.1 install and also enabled when I noticed my reboots. I have this disabled now (as I don't want that feature I'd like to keep it always disabled) and will check if that might have been the reason for the reboots.)
